features/steps/reusable_steps.rb

```diff
@@ -10,8 +10,18 @@
   step(step_text) if platform.downcase == Maze::Helper.get_current_platform
 end
 
-Then(/^on (iOS|macOS) (\d+) and later, (.+)/) do |platform, version, step_text|
-  step(step_text) if platform.downcase == Maze::Helper.get_current_platform && Maze.config.os_version >= version
+Then(/^on (\w+) ([0-9.]+) and later, (.+)/) do |test_platform, test_version, step_text|
+  actual_platform = Maze::Helper.get_current_platform
+  actual_version = Maze.config.os_version
+
+  $logger.info "Detected actual platform/version: #{actual_platform} #{actual_version}"
+
+  unless test_platform.downcase == actual_platform && actual_version >= test_version.to_f
+    $logger.info "Skipping #{test_platform} #{test_version} check on #{actual_platform} #{actual_version}"
```

Comment:
This logging seems unnecessary, is it debug code accidentally left in? @twometresteve 

```suggestion
  unless test_platform.downcase == actual_platform && actual_version >= test_version.to_f
```

Reply:
I wanted it to be obvious what the scenario was actually testing and what it wasn't, because Cucumber turns it green whether a check was performed and passed or no check at all was performed.  I was trying to guard against the behaviour of `Maze::Helper.get_current_platform` changing subtly and undermining the entire step quietly.

Reply:
...although the one on L17 could be removed.

Reply:
It just feels very odd to have such a noisy step - generally they produce no output unless there's an error.

Maybe $logger.debug instead ?

Or going back to the `Then(/^on (iOS|macOS) ...` way of doing it ensures only known platform names can be specified.